sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/DataFeedLiveTests.cs

```diff
@@ -2158,5 +2171,13 @@ private void ValidateGenericDataSource(DataFeedSource dataSource, bool isAdmin)
                 }
             }
         }
+
+        private T GetAuthenticationInstance<T>(string authenticationType)
+        {
+            Type dataSourceType = typeof(T);
+            PropertyInfo staticProperty = dataSourceType.GetProperty(authenticationType);
+
+            return (T)staticProperty.GetValue(null);
+        }
```

Comment:
I hate using `Reflection`, but being able to get the authentication type like this in 3 lines of code makes it worth it.

We have the structs `DataSourceA.AuthenticationType`, `DataSourceB.AuthenticationType`, `DataSourceC.AuthenticationType`, and `DataSourceD.AuthenticationType`. They are all different types and have no polymorphic relation between them. The only thing we know is that their members have the same names.

So, if we pass `authenticationType = "Basic"`, we'll be able to get `DataSource*.AuthenticationType.Basic`, regardless of what the type `T` is (and as long as the member `Basic` exists).

If we don't use Reflection, we'll need 4 helper methods in our tests, each one with a switch that goes through every possible `AuthenticationType` member and makes a conversion (15 cases in total). It already feels bad, but the service team confirmed they intend to have tens of data feeds and credentials in the long term, so this would grow out of control.